text/0019-milestone-payload/0019-milestone-payload.md

```diff
@@ -0,0 +1,136 @@
+            <td>Index Number</td>
+            <td>uint32</td>
+            <td>The index number of the milestone.</td>
+          </tr>
+          <tr>
+            <td>Timestamp</td>
+            <td>uint64</td>
+            <td>The Unix timestamp at which the milestone was issued. The unix timestamp is specified in seconds.</td>
+          </tr>
+          <tr>
+            <td>Parent1</td>
+            <td>ByteArray[32]</td>
+            <td>The Message ID of the first <i>Message</i> referenced by the milestone.</td>
+          </tr>
+          <tr>
+            <td>Parent2</td>
+            <td>ByteArray[32]</td>
+            <td>The Message ID of the second <i>Message</i> referenced by the milestone.</td>
+          </tr>
+          <tr>
+            <td>Inclusion Merkle Root</td>
+            <td>ByteArray[64]</td>
+            <td>512-bit hash based on all of the not-ignored state-mutating transactions referenced by the milestone. (<a href="https://github.com/iotaledger/protocol-rfcs/blob/master/text/0012-milestone-merkle-validation/0012-milestone-merkle-validation.md">RFC-0012</a>)</td>
+          </tr>
+          <tr>
+            <td>Keys Count</td>
+            <td>uint8</td>
+            <td>Number of public keys entries.</td>
+          </tr>
+          <tr>
+            <td>Public Keys</td>
+            <td>Array&lt;ByteArray[32]&gt;</td>
+            <td>An array of public keys to validate the signatures. The keys must be in lexicographical order.</td>
+          </tr>
+        </table>
+      </details>
+    </td>
+  </tr>
+  <tr>
+    <td>Signatures Count</td>
+    <td>uint8</td>
+    <td>Number of signature entries. The number must match the field <code>Keys Count</code>.</td>
+  </tr>
+  <tr>
+    <td>Signatures</td>
+    <td>Array&lt;ByteArray[64]&gt;</td>
+    <td>An array of signatures signing the serialized <i>Milestone Essence</i>. The signatures must be in the same order as the specified public keys.</td>
+  </tr>
+</table>
+
+## Generation
+
+- Generate a new _Milestone Essence_ corresponding to the Coordinator milestone.
+- Transmit the serialized _Milestone Essence_ to the corresponding number of signature service providers.
+  - The signature provider service will sign the received serialized bytes as-is.
+  - The signature provider will serialize the signature bytes and return them to the Coordinator.
+- Fill the `Signatures` field of the milestone payload with the received signature bytes.
+- Generate a *Message* as defined in [RFC-0017 (draft)](https://github.com/GalRogozinski/protocol-rfcs/blob/message/text/0017-message/0017-message.md) using the same `Parent1` and `Parent2` for the created _Milestone Payload_.
+
+## Syntactical validation
+
+- `Parent1` and `Parent2` of the payload must match `Parent1` and `Parent2` of the encapsulating _Message_.
+- `Keys Count` must be at least the _Signature Threshold_ and at most the number of _Applicable Public Keys_ for the current milestone index.
+- Public keys:
```

Comment:
We need to add something important here: the keys **must** not repeat.

Reply:
Added an explicit line that the public keys must be unique